hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -56,6 +57,21 @@
      */
     boolean allPartitionsIndexed(int ownedPartitionCount);
 
+    /**
+     * Notifies the index that a partition update is about to begin. Could be caused be either
+     * partition add (e.g. migration from another member, dynamic index creation), or partition
+     * remove (e.g. migration to another member).
+     * <p>
+     * While in this state, the index cannot be queried by the SQL engine safely, because it
+     * will produce inconsistent results.
+     * <p>
+     * Internally this call increments the counter of active partition updates. The counter
```

Comment:
As mentioned in other comment, do we distinguish which index partition is not safe to query and which is safe to query?

Reply:
No, because this is not important for SQL - if there is at least one partition in the inconsistent state, we cannot query the whole index. As mentioned in the comment above, predicate API will require a separate fix.